src/gameboy/mod.rs

```diff
@@ -54,16 +58,16 @@ impl Gameboy {
         Ok(())
     }
 
-    fn handle_sdl2_events(&mut self) -> Result<(), String> {
-        let mut event_pump = self.sdl_context.event_pump()?;
+    fn handle_sdl2_events(&mut self) -> Result<(), io::Error> {
+        let mut event_pump = self.sdl_context.event_pump().unwrap();
         for event in event_pump.poll_iter() {
             match event {
                 Event::Quit {..}
                 | Event::KeyDown {
                     keycode: Some(Keycode::Escape),
                     repeat: false,
                     ..
-                } => return Err("quit requested".to_string()),
+                } => return Err(io::Error::from_raw_os_error(0)),
```

Comment:
What is error code 0? 

Also we might need to workshop error handling with io::Error. It looks like io::Error::from_raw_os_error would break cross platform capabilities since windows and linux are going to have different error codes. Personally, I would like this to work on linux/windows/macos. Ya know, so our buddies can try it out.

Reply:
yo fuck windows

Reply:
tbh I'm pretty annoyed with the way rust handles exiting

Reply:
Why would this break on Windows or MacOS?
https://doc.rust-lang.org/std/io/struct.Error.html#method.from_raw_os_error

Reply:
I see. For now I'm down to focus on Linux and then update later.